File: solutions/0035. Search Insert Position/0035.cs
// Time complexity: O(logn)
// Space complexity: O(1)
public class Solution {
    public int SearchInsert(int[] nums, int target) {
        int l = 0;
        int r = nums.Length;

        while(l < r) {
            int m = l + (r - l) / 2;
            if (nums[m] == target) {
                return m;
            }
            if (nums[m] < target) {
                l = m + 1;
            } else {
                r = m;
            }
        }
        return l;
    }
}

File: solutions/0035. Search Insert Position/0035.cpp
// Time complexity: O(logn)
// Space complexity: O(1)
class Solution {
public:
  int searchInsert(const vector<int>& nums, const int target) {
    int left = 0;
    int right = nums.size();

    while (left < right) {
      int mid = left + (right - left) / 2;

      if (nums[mid] == target)
        return mid;
      
      if (nums[mid] < target)
        left = mid + 1;
      else
        right = mid;
    }
    
    return left;
  }
};